### PR TITLE
Add 'native_sparse_array' and 'native_sparse_array_to_indices_values_…

### LINKED ISSUE #10703: Add Sparse Array Functions to Paddle Backend
Add [Sparse Array Functions](https://www.paddlepaddle.org.cn/documentation/docs/en/api/index\_en.html) to Paddle backend

\_

>Please keep in mind that the proper way to link an issue to this list is to comment "- [ ] #issue\_number" while the issue's title only includes the name of the function you've chosen.

\_

## Experimental

- [x] is\_native\_sparse\_array
- [ ] native\_sparse\_array
- [ ] native\_sparse\_array\_to\_indices\_values\_and\_shape

### PR DESCRIPTION
**Pull request by metric-space**
close #10703 

# Testing
seems like the testing of this is covered here: `ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_sparse_array.py`

### CODE REVIEW

**Comment by metric-space** [comment]
I think some tests are failing because of the lack of formats that are available in other non-paddle backends. Formats like `bsr` and `csc` at the moment isn't supported by paddle. Still thinking about how to go about this :thinking:

**Comment by metric-space** [comment]
Update: I believe I figured out a way around the paddle unsupported sparse formats. The remaining failures seem to do with this reproducible error (I do not know if bug or not)

``` python
import ivy
import paddle
ivy.set_backend('paddle')

a = ivy.array([1,2,3,4,5])
ivy.flatten(a)

```
the error is 
```
During the handling of the above exception, another exception occurred:

  File "<stdin>", line 1, in <module>
  File "/home/user/ivy/ivy/utils/exceptions.py", line 128, in _handle_exceptions
    raise ivy.utils.exceptions.IvyError(fn.__name__, str(e))

IvyError: paddle: flatten: The input x should be a Tensor
```
this creeps up in within the `SparseArray` class's `to_dense_array` method

The stack trace is pretty clear, wondering why the conversion to paddle's tensor type isn't happening. Still investigating

**Comment by MahmoudAshraf97** [comment]
hello @metric-space and sorry for the delay, the code in the previous comment should be working fine now

**Comment by metric-space** [comment]
@MahmoudAshraf97 hello there again, was hoping if you could give this PR a review please

File: ivy/functional/backends/paddle/experimental/sparse_array.py [hunk 1]
```diff
@@ -17,9 +32,70 @@ def native_sparse_array(
     values=None,
     dense_shape=None,
     format="coo",
-):
-    raise IvyNotImplementedException()
+    device=None,
+    requires_grad=False,
+    dtype=None,
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/paddle/experimental/sparse_array.py:37]:
```suggestion

```
please make sure the function signature is consistent across ivy

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Review by MahmoudAshraf97 — APPROVED** [review]

**Pull request merged**